staking/types/messages.go

```diff
@@ -18,16 +22,189 @@ type StakingMessage interface {
+	Amount            *big.Int           `json:"amount" yaml:"amount"`
+}
+
+// msgCreateValidatorJSON - struct for creating a new validator for JSON
+type msgCreateValidatorJSON struct {
+	Description       Description        `json:"description" yaml:"description"`
+	Commission        CommissionRates    `json:"commission" yaml:"commission"`
+	MinSelfDelegation *big.Int           `json:"min_self_delegation" yaml:"min_self_delegation"`
+	StakingAddress    common.Address     `json:"staking_address" yaml:"staking_address"`
+	ValidatingPubKey  shard.BlsPublicKey `json:"validating_pub_key" yaml:"validating_pub_key"`
+	Amount            *big.Int           `json:"amount" yaml:"amount"`
+}
+
+// NewMsgCreateValidator creates a new validator
+func NewMsgCreateValidator(
+	description Description, commission CommissionRates, minSelfDelegation *big.Int, stakingAddress common.Address, validatingPubKey shard.BlsPublicKey, amount *big.Int) MsgCreateValidator {
+
+	return MsgCreateValidator{
+		Description:       description,
+		Commission:        commission,
+		MinSelfDelegation: minSelfDelegation,
+		StakingAddress:    stakingAddress,
+		ValidatingPubKey:  validatingPubKey,
+		Amount:            amount,
+	}
 }
 
 // Type ...
 func (msg MsgCreateValidator) Type() string { return "create_validator" }
 
 // Signer ...
-func (msg MsgCreateValidator) Signer() common.Address { return msg.Address }
+func (msg MsgCreateValidator) Signer() common.Address { return msg.StakingAddress }
+
+// MarshalJSON implements the json.Marshaler interface to provide custom JSON
+// serialization of the MsgCreateValidator type.
+func (msg MsgCreateValidator) MarshalJSON() ([]byte, error) {
+	return json.Marshal(msgCreateValidatorJSON{
+		Description:       msg.Description,
+		Commission:        msg.Commission,
+		MinSelfDelegation: msg.MinSelfDelegation,
+		StakingAddress:    msg.StakingAddress,
+		ValidatingPubKey:  msg.ValidatingPubKey,
+		Amount:            msg.Amount,
+	})
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface to provide custom
+// JSON deserialization of the MsgCreateValidator type.
+func (msg *MsgCreateValidator) UnmarshalJSON(bz []byte) error {
+	var msgCreateValJSON msgCreateValidatorJSON
+	if err := json.Unmarshal(bz, &msgCreateValJSON); err != nil {
+		return err
+	}
+
+	msg.Description = msgCreateValJSON.Description
+	msg.Commission = msgCreateValJSON.Commission
+	msg.MinSelfDelegation = msgCreateValJSON.MinSelfDelegation
+	msg.StakingAddress = msgCreateValJSON.StakingAddress
+	msg.ValidatingPubKey = msgCreateValJSON.ValidatingPubKey
+	msg.Amount = msgCreateValJSON.Amount
+
+	return nil
+}
+
+// ValidateBasic quick validity check
+func (msg MsgCreateValidator) ValidateBasic() error {
+	// note that unmarshaling from bech32 ensures either empty or valid
+	if msg.StakingAddress.Big().Uint64() == 0 {
+		return errors.New("[CreateValidator] address is empty")
+	}
+	emptyKey := shard.BlsPublicKey{}
```

Comment:
pull out as plain var, no need to make each time

Reply:
Done.